### ISSUE TITLE
Question about gradient_accumulation_steps

### ISSUE THREAD

**Issue by yuzc19** [Lightning-AI/lit-gpt#496]
https://github.com/Lightning-AI/lit-gpt/blob/38e23b8148e625e759fb3d979e2f5e9a1f4a6931/pretrain/openwebtext.py#L36

I think gradient_accumulation_steps here should be batch_size // micro_batch_size // devices since every device processes gradient_accumulation_steps batches and then optimizes the parameter. Am I right?

**Comment by zzzhacker** [comment]
Hi @yuzc19 ,
I also wanted to know about this. Because It also depends on using DP(effective batch size = micro_batch_size * gradient_accumulation_steps) vs. DDP. (effective batch size = micro_batch_size * gradient_accumulation_steps * num_of_gpus). 
I find the reference doc about this [here](https://lightning.ai/docs/pytorch/stable/advanced/training_tricks.html).

**Comment by carmocca** [comment]
@yuzc19. The variable name is not explicit about it, but it's meant to be the per-device gradient accumulation steps, making it consistent with the current definition

@zzzhacker. The code uses DDP

**Comment by yuzc19** [comment]
@carmocca Thanks for the timely reply. So the final gradient accumulation steps should be this variable*num_of_gpus, right?

<img width="798" alt="Screenshot 2023-09-04 at 22 10 40" src="https://github.com/Lightning-AI/lit-gpt/assets/53684662/0635b008-718f-4fe8-b65f-9814c7c7e992">

**Comment by carmocca** [comment]
That's right
